base/shell/explorer/startctxmnu.cpp

```diff
@@ -185,7 +185,7 @@ class CStartMenuBtnCtxMenu :
         UINT uiCmdId = PtrToUlong(lpici->lpVerb);
         if (uiCmdId != 0)
         {
-            if ((uiCmdId >= ID_SHELL_CMD_FIRST) && (uiCmdId <= ID_SHELL_CMD_LAST))
+            if ((uiCmdId < ID_SHELL_CMD_PROPERTIES ) || (uiCmdId > ID_SHELL_CMD_RESTORE_ALL))
```

Comment:
This is wrong.
As you can observe below, `uiCmdId - ID_SHELL_CMD_FIRST` depends on the fact that it falls in this range.
If it would be a string, with your code you would subtract `ID_SHELL_CMD_FIRST` from the address of the string, and then try to use it as string again, which will crash.